cranelift-codegen/meta/src/isa/x86/instructions.rs

```diff
@@ -290,5 +290,22 @@ pub fn define(
         .operands_out(vec![a]),
     );
 
+    let Idx = &operand_doc("Idx", uimm8, "Lane index");
+    let x = &operand("x", TxN);
+    let a = &operand("a", &TxN.lane_of());
+
+    ig.push(
+        Inst::new(
+            "x86_pextr",
+            r#"
+        Extract lane ``Idx`` from ``x``.
+        The lane index, ``Idx``, is an immediate value, not an SSA value. It
+        must indicate a valid lane index for the type of ``x``.
```

Comment:
Can you precise how this behaves with respect to the upper bits in the result (zero extension), please?

Reply:
What do you think of my suggestion in https://github.com/CraneStation/cranelift/pull/943#discussion_r321371969?